api/envoy/extensions/geoip_providers/maxmind/v3/maxmind.proto

```diff
@@ -39,4 +40,16 @@ message MaxMindConfig {
   // Common provider configuration that specifies which geolocation headers will be populated with geolocation data.
   common.v3.CommonGeoipProviderConfig common_provider_config = 4
       [(validate.rules).message = {required: true}];
+
+  // Optional full file path to the config for country mapping overrides, e.g. ``/etc/country-mapping.conf``.
+  // The file is expected to have a ``.conf`` extension, and should be JSON with the following format:
+  //
+  // .. code-block::json
+  //   {
+  //     "country_code": "country_code" # comment
```

Comment:
Probably worth clarifying that it's not actually json because json doesn't support comments.

I might also suggest using `//` comments rather than `#` comments, adding an optional ignore_comments argument in the json loader (to be used [here](https://github.com/envoyproxy/envoy/blob/main/source/common/json/json_internal.cc#L728) as the argument to [sax_parse](https://json.nlohmann.me/api/basic_json/sax_parse/)) and supporting comments that way ([nlohmann's blurb about comments](https://json.nlohmann.me/features/comments/))

That would be better tested than the regex which would, for example, have a surprising outcome if someone were to try to use some sort of meta country codes like
```
{
  "#X": "boop"
}
```
(The hacky regex method as currently coded will transform this into invalid json.)

Reply:
seems the json is not rendering - im guessing the comment is why (https://storage.googleapis.com/envoy-pr/bdfa297/docs/api-v3/extensions/geoip_providers/maxmind/v3/maxmind.proto.html)

if the comment is necessary you can add `:force:` to the json code-block, otherwise maybe just remove the comment

also wondering if the json code-block needs a space - ie `.. code-block:: json`

Reply:
made some changes on the parser

Reply:
JSON doesnt support any comments - so this is still failing

as said above - if comments really _should_ be included we can force it - but its indicating that (most) json parsers are not going to work with comments included